elastic/kibana
# [Stack monitoring] adapt ES queries to updated document structure

### Summary

Part of https://github.com/elastic/kibana/issues/121282

Updates stack monitoring to read off `.monitoring-{product}-8-*` datastreams created by ES at startup. This change only covers the elasticsearch product.
The `_mb` archived datasets used by functional and api tests were updated to target the new indices (with [this script](https://gist.github.com/klacabane/0a68e88317de0a912fe41d98a54052cf)) and the elasticsearch tests updated to support datastreams. also manually tested and compared the views against the 7.16 `xpack.enabled` version with a single-node snapshot setup

small fixes were required in [the metricbeat module](https://github.com/elastic/beats/pull/29548) to get some data to load

### Testing
- build metricbeat with [this change](https://github.com/elastic/beats/pull/29548)
- start metricbeat with `elasticsearch-xpack` module enabled
- load SM UI, elasticsearch section should appear and every screen should display correct informations
